lib/request_started_on_middleware.rb

```diff
@@ -0,0 +1,57 @@
+# Add to config/application.rb:
+#
+#     config.middleware.use 'RequestStartedOnMiddleware'
+#
+class RequestStartedOnMiddleware
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    start_request(env['PATH_INFO'], Time.now.utc)
```

Comment:
Pedantic, but can we extract this `Time.now` value from the `env` instead of looking it up ourselves?  Does it makes sense to make another `Time.now` allocation, or should use what is there?

Honestly asking here though, because I am curious if you think there is an advantage or disadvantage to one or the other.

Reply:
Good question.  I was thinking you're right... maybe we should use the one in the headers but then I was looking at how newrelic pulls the times out of the header and it looks faster to do it the way I already have it.  Check it out [here](https://github.com/newrelic/rpm/blob/d4bbb12c627f84c6fc64d6c51389b4a48e76654f/lib/new_relic/agent/instrumentation/queue_time.rb#L30-L48)

Reply:
So, yeah, I think what I have is fine.